deps/rabbitmq_cli/lib/rabbitmq/cli/diagnostics/commands/check_if_any_deprecated_features_are_used_command.ex

```diff
@@ -0,0 +1,65 @@
+## This Source Code Form is subject to the terms of the Mozilla Public
+## License, v. 2.0. If a copy of the MPL was not distributed with this
+## file, You can obtain one at https://mozilla.org/MPL/2.0/.
+##
+## Copyright (c) 2007-2023 VMware, Inc. or its affiliates.  All rights reserved.
+
+defmodule RabbitMQ.CLI.Diagnostics.Commands.CheckIfAnyDeprecatedFeaturesAreUsedCommand do
+  alias RabbitMQ.CLI.Core.DocGuide
+
+  alias RabbitMQ.CLI.Diagnostics.Commands.{
+    CheckIfClusterHasClassicQueueMirroringPolicyCommand
+  }
+
+  @behaviour RabbitMQ.CLI.CommandBehaviour
+
+  def scopes(), do: [:ctl, :diagnostics]
+
+  use RabbitMQ.CLI.Core.MergesNoDefaults
+
+  def validate([_ | _] = args, _) when length(args) != 0,
+    do: {:validation_failure, :too_many_args}
+
+  def validate([], %{formatter: formatter}) do
+    case formatter do
+      "report" -> :ok
```

Comment:
We do not validate formatters like this anywhere else (except for the `report` command which is highly unusual in general).

I will refactor this to do what other health checks do (format the responses as JSON), since they all share the same result invariant.

Reply:
The idea may be to, just like in `rabbitmq-diagnostics report`, to combine output of multiple commands.

A compound health check is something we don't have. Providing some details in the output can be a good idea but sounds like all we need to do is to output a list of enabled features.